pkg/apis/resource/structured/namedresources/types.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package namedresources
+
+import "k8s.io/apimachinery/pkg/api/resource"
+
+// Resources is used in NodeResourceModel.
+type Resources struct {
+	Instances []Instance
+}
+
+type Instance struct {
+	// Name is unique identifier among all resource instances managed by
+	// the driver on the node. It must be a DNS subdomain.
+	Name string
+
+	// Attributes defines the attributes of this resource instance.
+	// The name of each attribute must be unique.
+	Attributes []Attribute
+}
+
+type Attribute struct {
+	// Name is unique identifier among all resource instances managed by
+	// the driver on the node. It must be a DNS subdomain.
+	Name string
+
+	AttributeValue
+}
+
+// AttributeValue must have one and only one field set.
+type AttributeValue struct {
+	QuantityValue    *resource.Quantity
+	BoolValue        *bool
+	IntValue         *int64
+	IntSliceValue    *IntSlice
+	StringValue      *string
+	StringSliceValue *StringSlice
+	// TODO: Version     *SemVersion
+}
+
+type IntSlice struct {
+	Ints []int64
+}
+
+type StringSlice struct {
+	Strings []string
+}
```

Comment:
This level of indirection isn't needed anymore, right?

Reply:
It is needed, because of the [code generator issue for `*[]int64` and the `nil` vs. `[]int64{}` handing with `omitempty`](https://github.com/kubernetes/kubernetes/pull/123516#discussion_r1505819352).

Reply:
OK. I guess there were two different issues: one you determined to be an error in our test, but this one is different.